File: Lecture 3/FinalCode - Java/app/src/main/java/com/inovaeg/androidtrainingpharos2020/finaldemo/NetflixMovieOverviewActivity.java
package com.inovaeg.androidtrainingpharos2020.finaldemo;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import com.inovaeg.androidtrainingpharos2020.R;

public class NetflixMovieOverviewActivity extends AppCompatActivity {

    private Boolean isAddedToMyList = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_netflix_movie_overview);

        final Button myListButton = findViewById(R.id.btn_movie_overview_my_list);

        myListButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                isAddedToMyList = !isAddedToMyList;
                int drawableId;
                if (isAddedToMyList)
                    drawableId = R.drawable.ic_check;
                else
                    drawableId = R.drawable.ic_add;

                myListButton.setCompoundDrawablesWithIntrinsicBounds(null,
                        ContextCompat.getDrawable(NetflixMovieOverviewActivity.this,
                                drawableId), null, null);

            }
        });
    }
}

File: Lecture 3/FinalCode - Kotlin/app/src/main/java/com/inovaeg/androidtrainingpharos2020/finaldemo/NetflixMovieOverviewActivity.kt
package com.inovaeg.androidtrainingpharos2020.finaldemo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import com.inovaeg.androidtrainingpharos2020.R
import com.inovaeg.androidtrainingpharos2020.finaldemo.NetflixMovieOverviewActivity
import kotlinx.android.synthetic.main.activity_netflix_movie_overview.*

class NetflixMovieOverviewActivity : AppCompatActivity() {
    private var isAddedToMyList = false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_netflix_movie_overview)

        btn_movie_overview_my_list.setOnClickListener {
            isAddedToMyList = !isAddedToMyList
            val drawableId = if (isAddedToMyList) R.drawable.ic_check else R.drawable.ic_add
            btn_movie_overview_my_list.setCompoundDrawablesWithIntrinsicBounds(null,
                    ContextCompat.getDrawable(this@NetflixMovieOverviewActivity,
                            drawableId), null, null)
        }
    }
}